tests/test_coco_format.py

```diff
@@ -836,6 +847,320 @@ def test_can_pickle(self):
                 compare_datasets_strict(self, source, parsed)
 
 
+class CocoExtractorTests(TestCase):
+    @mark_requirement(Requirements.DATUM_ERROR_REPORTING)
+    def test_can_report_unexpected_file(self):
+        with TestDir() as test_dir:
+            with self.assertRaisesRegex(DatasetImportError, "JSON file"):
+                CocoInstancesExtractor(test_dir)
+
+    @mark_requirement(Requirements.DATUM_ERROR_REPORTING)
+    def test_can_report_missing_item_field(self):
+        for field in ["id", "file_name"]:
+            with self.subTest(field=field):
+                with TestDir() as test_dir:
+                    ann_path = osp.join(test_dir, "ann.json")
+                    anns = {
+                        "images": [
+                            {
+                                "id": 5,
+                                "width": 10,
+                                "height": 5,
+                                "file_name": "a.jpg",
+                            }
+                        ],
+                        "annotations": [],
+                        "categories": [],
+                    }
+                    anns["images"][0].pop(field)
```

Comment:
FWIW, `pop` is used to remove a key and return the corresponding value. When you just need to remove a key, the idiomatic way to do that is `del`.

Reply:
Maybe, but I find `del anns["images"][0][field]` really confusing. Are we going to delete a key or call the d-tor on the value?

Reply:
I don't know why you would think it's the latter; Python doesn't have any specific syntax for calling destructors. But it's just a nitpick, anyway, so I'll leave it up to you.